frontend/src/contracts.ts

```diff
@@ -230,88 +233,69 @@ async function setUpStakingContracts(web3: Web3) {
 }
 
 export async function setUpContracts(web3: Web3): Promise<Contracts> {
-
   const stakingContracts = await setUpStakingContracts(web3);
 
   const cryptoBladesContractAddr = getConfigValue('VUE_APP_CRYPTOBLADES_CONTRACT_ADDRESS') || (cryptoBladesNetworks as Networks)[networkId]!.address;
   const CryptoBlades = new web3.eth.Contract(cryptoBladesAbi as Abi, cryptoBladesContractAddr);
-  const [charactersAddr, weaponsAddr, randomsAddr, blacksmithAddr] = await Promise.all([
-    CryptoBlades.methods.characters().call(),
-    CryptoBlades.methods.weapons().call(),
-    CryptoBlades.methods.randoms().call(),
-    CryptoBlades.methods.blacksmith().call(),
-  ]);
+
+  const randomsAddr = await CryptoBlades.methods.randoms().call();
+  const charactersAddr = (charactersNetworks as Networks)[networkId]!.address;
+  const weaponsAddr = (weaponsNetworks as Networks)[networkId]!.address;
+  const blacksmithAddr = (blacksmithNetworks as Networks)[networkId]!.address;
+
 
   const Randoms = new web3.eth.Contract(randomsAbi as Abi, randomsAddr);
   const Characters = new web3.eth.Contract(charactersAbi as Abi, charactersAddr);
   const Weapons = new web3.eth.Contract(weaponsAbi as Abi, weaponsAddr);
   const Blacksmith = new web3.eth.Contract(blacksmithAbi as Abi, blacksmithAddr);
 
-  let SpecialWeaponsManager;
-  let specialWeaponsManagerAddr;
-  try {
-    specialWeaponsManagerAddr = await CryptoBlades.methods.specialWeaponsManager().call();
-  }
-  catch(e) {
-    console.error(e);
-  }
-
-  if(specialWeaponsManagerAddr) {
-    SpecialWeaponsManager = new web3.eth.Contract(specialWeaponsManagerAbi as Abi, specialWeaponsManagerAddr);
-  }
+  const specialWeaponsManagerAddr = (specialWeaponsManagerNetworks as Networks)[networkId]!.address;
+  const SpecialWeaponsManager = new web3.eth.Contract(specialWeaponsManagerAbi as Abi, specialWeaponsManagerAddr);
```

Comment:
What will happen when specialWeaponsManagerAddr is not there? From what I see, the previous solution guards it from throwing an error. We need to have it bulletproof, because iirc the missing address and creating a contract from it made app unusable